homeassistant/components/wiz/light.py

```diff
@@ -0,0 +1,358 @@
+            return 0
+        # If bulbtype is TW or RGB then return the kelvin value
+        try:
+            return color_utils.color_temperature_kelvin_to_mired(
+                self._bulbtype.kelvin_range.min
+            )
+        except WizLightNotKnownBulb:
+            _LOGGER.debug("Kelvin is not present in the library. Fallback to 2200")
+            return color_utils.color_temperature_kelvin_to_mired(2200)
+
+    def get_supported_features(self) -> int:
+        """Flag supported features."""
+        if not self._bulbtype:
+            # fallback
+            return SUPPORT_FEATURES_RGB
+        features = 0
+        try:
+            # Map features for better reading
+            if self._bulbtype.features.brightness:
+                features = features | SUPPORT_BRIGHTNESS
+            if self._bulbtype.features.color:
+                features = features | SUPPORT_COLOR
+            if self._bulbtype.features.effect:
+                features = features | SUPPORT_EFFECT
+            if self._bulbtype.features.color_tmp:
+                features = features | SUPPORT_COLOR_TEMP
+            return features
+        except WizLightNotKnownBulb:
+            _LOGGER.warning(
+                "Bulb is not present in the library. Fallback to full feature"
+            )
+            return SUPPORT_FEATURES_RGB
+
+    @property
+    def effect(self):
+        """Return the current effect."""
+        return self._effect
+
+    @property
+    def effect_list(self):
+        """Return the list of supported effects.
+
+        URL: https://docs.pro.wizconnected.com/#light-modes
+        """
+        return self._scenes
+
+    @property
+    def available(self):
+        """Return if light is available."""
+        return self._available
+
+    async def async_update(self):
+        """Fetch new state data for this light."""
+        await self.update_state()
+
+        if self._state is not None and self._state is not False:
+            self.update_brightness()
+            self.update_temperature()
+            self.update_color()
+            self.update_effect()
+
+    @property
+    def device_info(self):
+        """Get device specific attributes."""
+        return {
+            "connections": {(CONNECTION_NETWORK_MAC, self._mac)},
+            "name": self._attr_name,
+            "manufacturer": "WiZ Light Platform",
+            "model": f"{self._bulbtype.name} Mac: {self._mac}",
+        }
+
+    def update_state_available(self):
+        """Update the state if bulb is available."""
+        self._state = self._light.status
+        self._available = True
+
+    def update_state_unavailable(self):
+        """Update the state if bulb is unavailable."""
+        self._state = False
+        self._available = False
+
+    async def update_state(self):
+        """Update the state."""
+        try:
+            await self._light.updateState()
+        except (TimeoutError, WizLightTimeOutError) as ex:
```

Comment:
```suggestion
        except (ConnectionRefusedError, TimeoutError, WizLightTimeOutError) as ex:
```

based on testing:

```
   asyncio.create_task(stream.send(bytes(message, "utf-8")))
  File "/home/frenck/.pyenv/versions/3.9.9/lib/python3.9/asyncio/tasks.py", line 361, in create_task
    task = loop.create_task(coro)
Traceback (most recent call last):
  File "/home/frenck/.pyenv/versions/3.9.9/envs/home-assistant-core/lib/python3.9/site-packages/asyncio_dgram/aio.py", line 152, in send
    await super()._send(data)
  File "/home/frenck/.pyenv/versions/3.9.9/envs/home-assistant-core/lib/python3.9/site-packages/asyncio_dgram/aio.py", line 106, in _send
    _ = self.exception
  File "/home/frenck/.pyenv/versions/3.9.9/envs/home-assistant-core/lib/python3.9/site-packages/asyncio_dgram/aio.py", line 59, in exception
    raise exc
  File "/home/frenck/.pyenv/versions/3.9.9/lib/python3.9/asyncio/selector_events.py", line 1019, in _read_ready
    data, addr = self._sock.recvfrom(self.max_size)
ConnectionRefusedError: [Errno 111] Connection refused

```

(unplugged/replugged the bulb in bad timing)